llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp

```diff
@@ -946,14 +946,14 @@ bool AMDGPUTargetLowering::isFAbsFree(EVT VT) const {
 
   // Packed operations do not have a fabs modifier.
   return VT == MVT::f32 || VT == MVT::f64 ||
-         (Subtarget->has16BitInsts() && VT == MVT::f16);
+         (Subtarget->has16BitInsts() && (VT == MVT::f16 || VT == MVT::bf16));
 }
 
 bool AMDGPUTargetLowering::isFNegFree(EVT VT) const {
```

Comment:
It practically means SDag will not try to replace with with bit operation and live it for selection.

Reply:
I've been meaning to remove the check in that case, instcombine now does that

Reply:
Do any of the bfloat instructions have source modifiers?

Reply:
These are not different from f16, so yes.

Reply:
> I've been meaning to remove the check in that case, instcombine now does that

It would be very unfortunate not to let fneg to survive until the selection.